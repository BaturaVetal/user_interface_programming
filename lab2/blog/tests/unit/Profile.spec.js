import { shallowMount } from '@vue/test-utils';
import ProfileComponent from '@/components/Profile.vue';


describe('Profile.vue', () => {
  it('renders user profile information', () => {
    const wrapper = shallowMount(ProfileComponent, {
      data() {
        return {
          user: { name: 'Batura Vitaliy', email: 'test@example.com', gender: 'Чоловік', dob: '2000-01-01' }
        };
      }
    });
    expect(wrapper.text()).toContain('Ім’я: Batura Vitaliy');
    expect(wrapper.text()).toContain('Email: test@example.com');
  });


  it('handles missing user data gracefully', () => {
    const wrapper = shallowMount(ProfileComponent, {
      data() {
        return {
          user: null
        };
      }
    });
    expect(wrapper.text()).toContain('Інформація про користувача недоступна');
  });


  it('matches the snapshot', () => {
    const wrapper = shallowMount(ProfileComponent, {
      data() {
        return {
          user: { name: 'Batura Vitaliy', email: 'test@example.com', gender: 'Чоловік', dob: '2000-01-01' }
        };
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
