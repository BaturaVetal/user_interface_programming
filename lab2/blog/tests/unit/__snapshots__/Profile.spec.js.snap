// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile.vue matches the snapshot 1`] = `
<v-card>
  <v-card-title>
    Профіль користувача
  </v-card-title>
  <v-card-text>
    <v-list>
      <v-list-item>
        <v-list-item-content>
          Ім’я: Batura Vitaliy
        </v-list-item-content>
      </v-list-item>
      <v-list-item>
        <v-list-item-content>
          Email: test@example.com
        </v-list-item-content>
      </v-list-item>
      <v-list-item>
        <v-list-item-content>
          Стать: Чоловік
        </v-list-item-content>
      </v-list-item>
      <v-list-item>
        <v-list-item-content>
          Дата народження: 2000-01-01
        </v-list-item-content>
      </v-list-item>
      <!--v-if-->
    </v-list>
  </v-card-text>
</v-card>
`;
